apps/docs/content/guides/platform/migrating-and-upgrading-projects.mdx

```diff
@@ -93,6 +93,48 @@ If you upgrade to a paid plan while your project is paused, any expired one-clic
   src="/docs/img/guides/platform/paused-paid-tier.png"
 />
 
+#### Restoring a Downloaded Backup Locally
+
+Downloaded backups are plaintext SQL files that can be loaded into a Postgres instance using [psql](https://www.postgresql.org/docs/current/app-psql.html). You can use our open source build tooling if you would like to restore a backup into a local environment so you can interact with it. Here are the steps:
+
+Install the [nix](https://nixos.org/):
+
+```sh
+curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix | sh -s -- install
+```
+
+Next, we need to start postgres locally.
+
+If you know the project's Postgres version from your project's [infrastructure project settings](https://supabase.com/dashboard/project/_/settings/infrastructure) page, you can use it to spin up an identical image locally. Using an identical image improves the odds of successfully restoring your project because all of the extension versions and configuration will match exactly.
+
+Given Postgres version `15.6.1.121`, start postgres locally with the nix command:
+
+```sh
+nix run 'github:supabase/postgres?ref=15.6.1.127#start-server' 15
+```
+
+where `15` refers to the backup's Postgres major version.
+
+Note that the earliest Supabase Postgres version that supports a direct reference is `15.6.1.115`. If you don't know your project's Postgres version or it is lower than `15.6.1.115` then omit the `ref` option to build our most recent image:
```

Comment:
"To build the most recent image"

Also, do we have tooling for the most recent /of a major postgres version/ option? Seems more likely to work, even if inexact, if it's at least the same major pg version. But also folks would need to know, or we'd need to include that data in the backup.

Reply:
The dump file contains the db version:

```sql
-- Dumped from database version 15.6
-- Dumped by pg_dump version 15.8 (Debian 15.8-1.pgdg120+1)
```

Although not right at the top, but could still be searched.

Reply:
`15.6.1.115` is a git tag. If, at that git tag we are producing images for 15 and 16 you can pass
```
nix run 'github:supabase/postgres?ref=15.6.1.115#start-server' 15
```
to start pg 15 and 
```
nix run 'github:supabase/postgres?ref=15.6.1.115#start-server' 16
```
to start 16.

Once we drop pg 15 support we'll take it out of the build, but that won't be until 100% of projects have moved off 15 because we need to keep the build around for spinning up new read replicas.

Once 15 is fully removed I'll come back to these docs and add a "if your image is pg15 the maximum git tag you can use is ..." just like we specify the minimum currently